### PR TITLE
Write navigation docs

### PR DESCRIPTION
**Pull request by tobias-mintlify**
## Summary

We had not written docs focusing on the `navigation` field. I've added a separate page so new users can learn how to add pages without understanding all the settings in detail. Most importantly, I've added a warning we cannot support the `/api/` folder because of Next.js

## Test Plan

Run `mintlify dev` to preview the docs.

### CODE REVIEW

File: docs/mint.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "navigation": [
     {
       "group": "Getting Started",
-      "pages": ["quickstart", "development"]
+      "pages": ["quickstart", "development", "navigation"]
```
Comment (hanywang2) [inline docs/mint.json:48]: I think it also makes more sense to move it to the "Settings" group

File: docs/navigation.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        "group": "Design Principles",
+        "pages": [
+            "overview",
+            {
+                "group": "Nested Reference Pages",
+                "pages": ["nested-page-name"]
+            }
+        ]
+    }
```
Comment (hahnbeelee) [inline docs/navigation.mdx:32]: Keep this consistent with the regular one, so it's easy to see the differences between the two.
```suggestion
"navigation": [
    {
        "group": "Getting Started",
        "pages": [
            "quickstart",
            {
                "group": "CLI",
                "pages": ["cli/overview", "cli-commands"]
            }
        ]
    }
```

File: docs/navigation.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+]
+```
+
+</CodeGroup>
+
+## Folders
+
+Simply put your `.mdx` files in folders and update the paths in `mint.json`.
```
Comment (hahnbeelee) [inline docs/navigation.mdx:40]:
```suggestion
Simply put your MDX files in folders and update the paths in `mint.json`.
```

File: docs/navigation.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+For example, `https://yoursite.com/my-folder/my-page` requires you to have a folder called `my-folder` containing an mdx file called `my-page.mdx`.
```
Comment (hahnbeelee) [inline docs/navigation.mdx:42]: Nit: Capitalize MDX

File: docs/navigation.mdx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        "group": "Group Name",
+        "pages": ["my-folder/my-page"]
+    }
+]
+```
+
+## Hidden Pages
+
+Mdx files you don't include in `mint.json` will still show up on your website. Hidden pages never show up in the sidebar navigation but can be found through the search bar.
```
Comment (hahnbeelee) [inline docs/navigation.mdx:65]:
```suggestion
MDX files you don't include in `mint.json` will not be included in the sidebar, but are accessible by navigating directly to the path or through the search bar.
```

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
This also seems like a good place to talk about anchors

**Review by hahnbeelee — APPROVED** [review]

**Pull request merged**